src/main/java/Duke/TaskList.java

```diff
@@ -0,0 +1,166 @@
+     *
+     * @param i index of the current task in the tasklist
+     * @return Marked Tasks.Task
+     */
+    public Task markTask(int i) {
+        Task t = tasks.get(i - 1);
+        t.setMarked();
+        return t;
+    }
+
+    /**
+     * Unmarks the specified task
+     *
+     * @param i
+     * @return Unmarked Tasks.Task
+     */
+    public Task unmarkTask(int i) {
+        Task t = tasks.get(i - 1);
+        t.setUnmarked();
+        return t;
+    }
+
+    /**
+     * Returns a list of tasks with the given task name
+     *
+     * @param taskName string of task
+     * @return arraylist of current tasks
+     */
+    public ArrayList<Task> findTask(String taskName) {
+        ArrayList<Task> foundTasks = new ArrayList<Task>();
+        for (Task t : tasks) {
+            if (t.getTaskName().contains(taskName)) {
+                foundTasks.add(t);
+            }
+        }
+        return foundTasks;
+    }
+
+    /**
+     * Gets the number of tasks in the taskList
+     *
+     * @return size of taskList
+     */
+    public int getSize() {
+        return tasks.size();
+    }
+
+    /**
+     * Prints the tasks into the taskList in the correct format
+     *
+     * @return string that prints the tasks
+     */
+    public String printList() {
+        String str = "";
+        try {
+            if (tasks.size() == 0) {
+                throw new DukeException(Constants.EMPTY_LIST);
+            }
+            else {
+                str += (Constants.LIST);
+                for (int i = 0; i < tasks.size(); i++) {
+                    str += (String.format("%d.%s \n", i + 1, tasks.get(i).toString()));
+                }
+            }
+        }
+        catch (DukeException e) {
+            System.out.println(e.getMessage());
+        }
+        return str;
+    }
+
+    /**
+     * Prints the tasks in the given taskList in the correct format
+     *
+     * @return string that prints the tasks
+     */
+    public String printList(ArrayList<Task> givenTasks) {
+        String str = "";
+        try {
+            if (givenTasks.size() == 0) {
+                throw new DukeException(Constants.EMPTY_LIST);
+            }
+            else {
+                str += (Constants.FOUND_LIST);
+                for (int i = 0; i < givenTasks.size(); i++) {
+                    str += String.format("%d.%s \n", i + 1, givenTasks.get(i).toString());
+                }
+            }
+        }
+        catch (DukeException e) {
+            System.out.println(e.getMessage());
+        }
+        return str;
+    }
+
+    /**
+     * Puts all the tasks in taskList into a string for storing into a file
+     *
+     * @return ArrayList containing all the tasks in Strings
+     */
+    public ArrayList<String> tasksToString() {
+        ArrayList<String> tasksStrings = new ArrayList<String>();
+        for (int i = 0; i < tasks.size(); i++) {
+            Task t = tasks.get(i);
+            String type = t.toString().substring(1,2);
```

Comment:
Maybe add a small space between numbers and commas?
```suggestion
            String type = t.toString().substring(1, 2);
```

Reply:
oops, thanks for highlighting 👍